File: greedy/leetcode_455/solution.js
/**
 * @param {number[]} g
 * @param {number[]} s
 * @return {number}
 */
var findContentChildren = function(g, s) {
    let greedIndex = 0;
    let cookieIndex = 0;

    g = new Uint32Array(g).sort();
    s = new Uint32Array(s).sort();

    while (greedIndex < g.length && cookieIndex < s.length) {
        if (s[cookieIndex] >= g[greedIndex]) {
            greedIndex ++;
            cookieIndex ++;
        } else {
            cookieIndex ++;
        }
    }

    return greedIndex;
};

File: greedy/leetcode_455/solution.py
from typing import List

class Solution:
    def findContentChildren(self, g: List[int], s: List[int]) -> int:
        greed_index = 0
        cookie_index = 0

        g.sort()
        s.sort()

        while greed_index < len(g) and cookie_index < len(s):
            if s[cookie_index] >= g[greed_index]:
                greed_index += 1
                cookie_index += 1
            else:
                cookie_index += 1
        
        return greed_index